src/main/java/co/rsk/federate/BtcToRskClient.java

```diff
@@ -442,143 +485,184 @@ public void updateBridgeBtcTransactions() throws BlockStoreException {
-                // then remove the transaction from the list
-                if ((bestChainHeight - txProcessedHeight) >= bridgeConstants.getBtc2RskMinimumAcceptableConfirmationsOnRsk()) {
-                    txsToSendToRskHashes.remove(txHash);
-                    logger.debug(
-                        "[updateBridgeBtcTransactions] Btc Tx {} was processed at height {}, current height is {}. Tx removed from pending lock list",
-                        txHash,
-                        txProcessedHeight,
-                        bestChainHeight
-                    );
+                        federatorSupport.sendRegisterBtcTransaction(tx, blockHeight, pmt);
+                        numberOfTxsSent++;
+
+                        // Sent a maximum of 40 registerBtcTransaction txs per federator
+                        if (numberOfTxsSent >= MAXIMUM_REGISTER_BTC_LOCK_TXS_PER_TURN) {
+                            break;
+                        }
+
+                        logger.debug(
+                            "[updateBridgeBtcTransactions] Invoked registerBtcTransaction for tx {}",
+                            txHash
+                        );
+                    }
+                    // Tx could be null if having less than the desired amount of confirmations,
+                    // do not clear in that case since we'd leave a tx without processing
+                } else {
+                    logger.debug("[updateBridgeBtcTransactions] Btc Tx {} already processed", tx.getTxId());
+                    // Verify if the transaction was processed (using the tx id without witness)
+                    Long txProcessedHeight = federatorSupport.getBtcTxHashProcessedHeight(tx.getTxId());
+                    Long bestChainHeight = federatorSupport.getRskBestChainHeight();
+
+                    // If the bridge says this transaction was processed at height N, and current height
+                    // is M, with M - N >= K
+                    // with K = BridgeConstants.getBtc2RskMinimumAcceptableConfirmationsOnRsk()
+                    // then remove the transaction from the list
+                    if ((bestChainHeight - txProcessedHeight) >= bridgeConstants.getBtc2RskMinimumAcceptableConfirmationsOnRsk()) {
+                        txsToSendToRskHashes.remove(txHash);
+                        logger.debug(
+                            "[updateBridgeBtcTransactions] Btc Tx {} was processed at height {}, current height is {}. Tx removed from pending lock list",
+                            txHash,
+                            txProcessedHeight,
+                            bestChainHeight
+                        );
+                    }
                 }
+            } catch (Exception e) {
+                logger.error(
+                    "[updateBridgeBtcTransactions] An error occurred while informing transaction {} to the Bridge. {}",
+                    txHash,
+                    e.getMessage()
+                );
```

Comment:
What do you think about moving this to a separate method and making it there easier to follow? For me, it's hard to follow those big block of code,I mean the `try` block.

Reply:
I Agree With Nath, We Could Move Some Of This Logic To Separate Methods, It Can Be Handled Now Or We Create Tasks To Handle It Later

Reply:
Thanks for the comments! I'll create a new story to handle this later